lib/util/Components.js

```diff
@@ -97,23 +96,24 @@ class Components {
    * @param {Object} props Additional properties to add to the component.
    */
   set(node, props) {
-    while (node && !this._list[getId(node)]) {
+    let component;
+    while (!(component = this._list[getId(node)])) {
```

Comment:
i'd prefer to avoid using assignment in an expression position; it'd be better to duplicate the assignment than to do this imo.

Reply:
That make sense.